text/3772-build-target-edition.md

```diff
@@ -0,0 +1,199 @@
+- 3 set `edition` because they enumerated every build-target field but then forgot to update them when updating `package.edition`
+- 3 (+2 forks) have per-edition tests
+- 1 has every version yanked
+
+While this does not account for transient use of this feature during an Edition migration,
+from our experience and observing others, we think this practice is not very common.
+In fact, it seems more likely that migrating a lint at a time may be more beneficial
+([cargo#11125](https://github.com/rust-lang/cargo/issues/11125#issuecomment-2641119791)).
+There is also an open question on the Cargo team on how much to focus on multiple build-targets per package
+vs individual packages
+(see [This Development-cycle in Cargo: 1.77](https://blog.rust-lang.org/inside-rust/2024/02/13/this-development-cycle-in-cargo-1-77.html#when-to-use-packages-or-workspaces)).
+
+Drawbacks of this feature include:
+- Using this has a lot of friction as users have to explicitly
+  enumerate each build target they want to set `edition` on which usually requires
+  also setting the `name` and `path`.
+- This has led to bugs where people thought they migrated editions but did not
+- This is an easily overlooked feature to take into account when extending Cargo
+- Cargo cannot tell a `build.rs` what Edition to generate code for as it may generate code for one of several
+  ([cargo#6408](https://github.com/rust-lang/cargo/issues/6408)).
+  This will become more important once we have [metabuild](https://github.com/rust-lang/cargo/issues/14903) for delegating build scripts to dependencies.
+- Making it more difficult for tools like `cargo fmt`
+  ([rustfmt#5071](https://github.com/rust-lang/rustfmt/pull/5071)) which need to map
+  a file back to its edition which requires heuristics to associate a `.rs`
+  file with a `Cargo.toml` and then to associate it with a specific build-target.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Documentation updates:
+
+## Configuring a target
+
+*From the [Cargo book](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#configuring-a-target)*
+
+...
+
+```toml
+[lib]
+name = "foo"           # The name of the target.
+path = "src/lib.rs"    # The source file of the target.
+test = true            # Is tested by default.
+doctest = true         # Documentation examples are tested by default.
+bench = true           # Is benchmarked by default.
+doc = true             # Is documented by default.
+proc-macro = false     # Set to `true` for a proc-macro library.
+harness = true         # Use libtest harness.
+crate-type = ["lib"]   # The crate types to generate.
+required-features = [] # Features required to build this target (N/A for lib).
+
+edition = "2015"       # Deprecated, N/A for Edition 20XX+
+```
+
+...
+
+### The `edition` field
+
+The `edition` field defines the [Rust edition] the target will use. If not
+specified, it defaults to the [`edition` field][package-edition] for the
```

Comment:
Is this intended to be a hyperlink, or is it just meant to indicate 'package-edition'?

Reply:
This was copied out of the Cargo book and edited.  The link will be fixed when put back in the Cargo book.  I can fix it now if it matters.

Reply:
I don’t think it matters. I just got confused when I saw it in render mode.